src/osdatahub/NamesAPI/names_api.py

```diff
@@ -0,0 +1,160 @@
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+        data = GrowList()
+        if not all([str(p).isnumeric() for p in point]):
+            raise TypeError("All values in argument \"point\" must be numeric")
+        if not 0.01 <= radius <= 1000:
+            raise ValueError(f"Argument \"radius\" must be between 0 and 1000, but had value {radius}")
+
+        params = {"point": ",".join([str(c) for c in point]), "radius": radius}
+        if local_type:
+            params.update({"fq": self.__format_fq(local_type=local_type)})
+        try:
+            response = requests.get(self.__endpoint("nearest"), params=params)
+            data.extend(self.__format_response(response))
+        except KeyError:
+            response.raise_for_status()
+        return addresses_to_geojson(data.values, crs="EPSG:27700")
+
+    @staticmethod
+    @typechecked
+    def __format_fq(bbox_filter: Extent = None, local_type: Union[str, Iterable] = None) -> list:
+        """
+        Formats optional fq arguments for Names API query
+
+        Args:
+            bbox_filter (Extent, optional): Filters the results to a certain area. Must be British National Grid
+                (EPSG:27700) CRS
+            local_type (Union[str, Iterable], optional): Filters the results to certain local types. Available local
+                types can be found at the bottom of https://osdatahub.os.uk/docs/names/technicalSpecification
+
+        Returns:
+            list of fq filtering arguments
+        """
+        fq_args = []
+        if local_type:
+            # check that all given local types are valid
+            print(local_type)
+            invalid_local_types = validate_local_type(local_type)
+            if invalid_local_types:
+                raise ValueError(f"The local type(s) {invalid_local_types} are not valid local types")
+            # builds local_type query whether given one argument or multiple
+            if isinstance(local_type, str):
+                local_types = "LOCAL_TYPE:" + get_local_type(local_type)
+            elif isinstance(local_type, Iterable):
+                local_types = " ".join([f"LOCAL_TYPE:{get_local_type(arg)}" for arg in local_type])
+            else:
+                raise TypeError(
+                    f"'local_type' argument must be Iterable or str, but was type {type(local_type)}")
+            fq_args.append(local_types)
+
+        # adds bbox filter to argument
+        if bbox_filter:
+            if bbox_filter.crs != "EPSG:27700":
+                raise ValueError("'bbox_filter' argument must have CRS of British National Grid (EPSG:27700). Its CRS "
+                                 f"is {bbox_filter.crs}")
+            fq_args.append("BBOX:" + str(bbox_filter.bbox.to_string()))
+
+        return fq_args
+
+    @staticmethod
+    def __format_response(response) -> list:
```

Comment:
Is it essential for this to be a static method?

Reply:
No, it could just as easily be outside the class. Would this be a better way of doing it?

Reply:
I don't think there's really a hard and fast rule but I've started putting static methods in a utils file because they reduce the cohesion of a class :)

Reply:
@JEPooleyOS how do you feel about this? I copied the static method function from the Features and Places API